pkg/kubelet/pleg/generic.go

```diff
@@ -395,9 +422,34 @@ func computeEvents(oldPod, newPod *kubecontainer.Pod, cid *kubecontainer.Contain
 	}
 	oldState := getContainerState(oldPod, cid)
 	newState := getContainerState(newPod, cid)
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && oldPod != nil && newPod != nil && cachedPodStatus != nil && podStatus != nil {
+		oldContainerStatus := cachedPodStatus.FindContainerStatusByContainerID(cid)
+		newContainerStatus := podStatus.FindContainerStatusByContainerID(cid)
+		if oldContainerStatus != nil && newContainerStatus != nil && !containerResourceSame(oldContainerStatus.Resources, newContainerStatus.Resources) {
+			klog.V(5).InfoS("resize pods triggers the plegContainerUnknown event", "oldContainerStatus", oldContainerStatus, "newContainerStatus", newContainerStatus)
+			return generateEvents(pid, cid.ID, oldState, plegContainerUnknown)
```

Comment:
It might be better to consider an edge case where a resized container has exited at the same time (newState==`exited`).